### PR TITLE
[Buttons] Add snapshot test for dynamic type

### PR DESCRIPTION
**Pull request by codeman7**
## Motivation
In order to capture the dynamic type visual changes within `MDCButton` for dynamic type we need to add test for the different size categories.

## Detailed changes
Add snapshot test for all `UIContentSizeCategory` values by overriding the `traitCollection` on a _fake_ MDCButton subclass. Currently these snapshots are the same because `MDCButton` uses the `mdc_scaledFontForCurrentSizeCategory` method within `MDCFontScaler` instead of `mdc_scaledFontForSizeCategory:`

### CODE REVIEW

File: components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+#import "MaterialSnapshot.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <UIKit/UIKit.h>
+
+#import "MaterialButtons.h"
+#import "MaterialTypography.h"
+
+/** A @c MDCButton test fake to override the @c traitCollection to test for dynamic type. */
+@interface MDCFakeDynamicTypeButton : MDCButton
```
Comment (romoore) [inline components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m]: Naming: `ButtonDynamicTypeSnapshotTestFakeButton`

File: components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  [view sizeToFit];
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+/**
+ Used to set the @c UIContentSizeCategory on an @c MDCButton.
+
+ @note On iOS 9 or below this method has now impact.
```
Comment (romoore) [inline components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m]: s/has now/has no/

**Review by romoore — APPROVED** [review]
Please update the PR summary (commit message) to document that these snapshots are capturing "broken" behavior.

File: components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m (hunk 1 above)
Comment (codeman7) [reply components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m]: Done

File: components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m (hunk 2 above)
Comment (codeman7) [reply components/Buttons/tests/snapshot/ButtonDynamicTypeSnapshotTests.m]: Done

**Pull request merged**